GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -657,13 +657,21 @@ private Workspace GetWorkspace(params WorkingFolder[] folders)
             var workspace = VersionControl.CreateWorkspace(GenerateWorkspaceName());
             try
             {
-                foreach (WorkingFolder folder in folders)
-                    workspace.CreateMapping(folder);
+                SetWorkspaceMappingFolders(workspace, folders);
             }
             catch (MappingConflictException e)
             {
-                TryToDeleteWorkspace(workspace);
-                throw new GitTfsException(e.Message).WithRecommendation("Run 'git tfs cleanup-workspaces' to remove the workspace.");
+                try
+                {
+                    foreach (WorkingFolder folder in folders)
+                        CleanupWorkspaces(folder.LocalItem);
+                    SetWorkspaceMappingFolders(workspace, folders);
```

Comment:
Ahh, I see. If `SetWorkspaceMappingFolders` succeeds this time, git-tfs will then make it to the `return workspace` line, which means success. :cool: